dagshub/data_engine/model/datasource.py

```diff
@@ -513,7 +513,34 @@ def func():
 
         return func()
 
-    def upload_metadata_from_dataframe(self, df: "pandas.DataFrame", path_column: Optional[Union[str, int]] = None):
+    def upload_metadata_from_file(self, file_path, path_column: Optional[Union[str, int]] = None, remote: bool = False):
+        """
+        Upload metadata from a file.
+
+        Args:
+            file_path: Path to the file with metadata. Allowed formats are CSV, Parquet, ZIP, GZ.
+            path_column: Column with the datapoints' paths. Can either be the name of the column, or its index.
+                If not specified, the first column is used.
+            remote: Whether to upload the metadata via remote or local upload. Default is False.
+                Remote upload means that the data engine API calls will be done from remote machine,
+                rather than from local machine which runs this client.
+        """
+        self.source.get_from_dagshub()
```

Comment:
Did you have to add this due to something that showed up during the testing?
In general this function should only be called _after_ uploading metadata (synchronously) to make sure that the field schema in the client is synchronized with the backend (for e.g. making sure the users filter on existing fields).

For this particular call this seems redundant